src/main/java/org/jboss/threads/QueuedViewExecutor.java

```diff
@@ -213,16 +213,20 @@ public String toString() {
             this.command = command;
         }
 
-        void interrupt() {
+        synchronized void interrupt() {
             final Thread thread = this.thread;
             if (thread != null) {
                 thread.interrupt();
             }
         }
 
         public void run() {
-            boolean removeFromAllWrappersAfterCompletion = true;
+            boolean resetStateOnCompletion = true;
             thread = Thread.currentThread();
+            // Interruption may be missed between when a TaskWrapper is submitted
+            // to the delegate executor, and when the task begins to execute.
+            // This must execute after thread is set.
+            if (state == ST_SHUTDOWN_INT_REQ) Thread.currentThread().interrupt();
```

Comment:
This would be more correct inside the retry loop. If the delegate has shut down and the view is terminated, it's possible a previous interruption was handled by a previous `command` in this wrapper, and must be set again for subsequent tasks.